public/pages/Dashboard/utils/tableUtils.js

```diff
@@ -164,65 +177,91 @@ export const alertColumns = (
+        // 2. If the monitor is created via visual editor, translate ui_metadata dsl filter to ppl filter
+        let formikToPPLFilters = [];
+        let pplBucketValue = 1;
+        let pplBucketUnitOfTime = 'HOUR';
+        const isVisualEditorMonitor = monitorDefinition?.ui_metadata?.search?.searchType === SEARCH_TYPE.GRAPH;
+        if (isVisualEditorMonitor) {
+          const uiFilters = monitorDefinition?.ui_metadata?.search?.filters || []
+          formikToPPLFilters = uiFilters.map((filter) => OPERATORS_PPL_QUERY_MAP[filter.operator].query(filter));
+          pplBucketValue = monitorDefinition?.ui_metadata?.search?.bucketValue || 1;
+          pplBucketUnitOfTime = BUCKET_UNIT_PPL_UNIT_MAP[monitorDefinition?.ui_metadata?.search?.bucketUnitOfTime] || 'HOUR';
+        }
         delete monitorDefinition.ui_metadata;
         delete monitorDefinition.data_sources;
 
+        // 3. get data triggers the alert and fetch log patterns
         let monitorDefinitionStr = JSON.stringify(monitorDefinition);
-
-        // 2. get data triggers the alert
         let alertTriggeredByValue = '';
         let dsl = '';
         let index = '';
+        let topNLogPatternData = '';
         if (
           monitorResp.resp.monitor_type === MONITOR_TYPE.QUERY_LEVEL ||
           monitorResp.resp.monitor_type === MONITOR_TYPE.BUCKET_LEVEL
         ) {
+          // 3.1 preprocess index, only support first index use case
           const search = monitorResp.resp.inputs[0].search;
-          const indices = String(search.indices);
-          const splitIndices = indices.split(',');
-          index = splitIndices.length > 0 ? splitIndices[0].trim() : '';
+          index = String(search.indices).split(',')[0]?.trim() || '';
+          // 3.2 preprocess dsl query with right time range
           let query = JSON.stringify(search.query);
           // Only keep the query part
           dsl = JSON.stringify({ query: search.query.query });
-          if (query.indexOf('{{period_end}}') !== -1) {
-            query = query.replaceAll('{{period_end}}', alert.start_time);
-            const alertStartTime = moment.utc(alert.start_time).format('YYYY-MM-DDTHH:mm:ss');
-            dsl = dsl.replaceAll('{{period_end}}', alertStartTime);
+          let latestAlertTriggerTime = '';
+          if (query.indexOf(PERIOD_END_PLACEHOLDER) !== -1) {
+            query = query.replaceAll(PERIOD_END_PLACEHOLDER, alert.last_notification_time);
+            latestAlertTriggerTime = moment.utc(alert.last_notification_time).format(DEFAULT_DSL_QUERY_DATE_FORMAT);
+            dsl = dsl.replaceAll(PERIOD_END_PLACEHOLDER, latestAlertTriggerTime);
             // as we changed the format, remove it
             dsl = dsl.replaceAll('"format":"epoch_millis",', '');
             monitorDefinitionStr = monitorDefinitionStr.replaceAll(
-              '{{period_end}}',
-              alertStartTime
+              PERIOD_END_PLACEHOLDER,
```

Comment:
To keep time format consistent in `monitorDefinitionStr`, how about replacing `alert.last_notification_time`, `alert.start_time` with their related format time as well? So LLM will never use unix timestamp in its response.

Reply:
Done. It's reflected in filtered active alerts.